projects/dot-rules/src/lib/semantic/modules/dropdown/dropdown.ts

```diff
@@ -1,163 +1,131 @@
-  @Input() allowAdditions: boolean;
-  @Input() minSelections: number;
-  @Input() maxSelections: number;
-
-  @Output() onDropDownChange: EventEmitter<any> = new EventEmitter();
-  @Output() touch: EventEmitter<any> = new EventEmitter();
-  @Output() enter: EventEmitter<boolean> = new EventEmitter(false);
-
-  modelValue: string;
-
-  private _optionsAry: InputOption[] = [];
-  private _options: BehaviorSubject<InputOption[]>;
-  private elementRef: ElementRef;
-
-  onChange: Function = () => {};
-  onTouched: Function = () => {};
-
-  constructor(
-    elementRef: ElementRef,
-    @Optional() control: NgControl,
-    private loggerService: LoggerService
-  ) {
-    if (control && !control.valueAccessor) {
-      control.valueAccessor = this;
+export class Dropdown implements ControlValueAccessor, OnChanges {
+    @Input()
+    set value(value: string) {
+        this.modelValue = value;
     }
-    this.placeholder = '';
-    this.allowAdditions = false;
-    this.minSelections = 0;
-    this.maxSelections = 1;
-    this._options = new BehaviorSubject(this._optionsAry);
-    this.elementRef = elementRef;
-  }
-
-  @Input()
-  set value(value: string) {
-    this.modelValue = value;
-  }
-
-  get options() {
-    return this._options;
-  }
-
-  writeValue(value: any): void {
-    this.modelValue = _.isEmpty(value) ? '' : value;
-  }
-
-  registerOnChange(fn): void {
-    this.onChange = fn;
-  }
-
-  registerOnTouched(fn): void {
-    this.onTouched = fn;
-  }
-
-  fireChange(value: any): void {
-    this.modelValue = value;
-    if (this.onDropDownChange) {
-      this.onDropDownChange.emit(value);
-      this.onChange(value);
-      this.fireTouch(value);
+    @Input() options: any;
+    @Input() name: string;
+    @Input() placeholder: string;
+    @Input() allowAdditions: boolean;
+    @Input() minSelections: number;
+    @Input() maxSelections: number;
+
+    @Output() onDropDownChange: EventEmitter<any> = new EventEmitter();
+    @Output() touch: EventEmitter<any> = new EventEmitter();
+    @Output() enter: EventEmitter<boolean> = new EventEmitter(false);
+
+    modelValue: string;
+    dropdownOptions: Observable<MenuItem[]>;
+
+    constructor(@Optional() control: NgControl) {
+        if (control && !control.valueAccessor) {
+            control.valueAccessor = this;
+        }
+        this.placeholder = '';
+        this.allowAdditions = false;
+        this.minSelections = 0;
+        this.maxSelections = 1;
     }
-  }
-
-  fireTouch($event): void {
-      this.onTouched();
-      this.touch.emit($event);
-
-  }
 
-  hasOption(option: InputOption): boolean {
-    const x = this._optionsAry.filter(opt => {
-      return option.value === opt.value;
-    });
-    return x.length !== 0;
-  }
+    ngOnChanges(changes: SimpleChanges) {
+        if (changes.options && changes.options.currentValue) {
+            this.dropdownOptions = from(this.options).pipe(
+                mergeMap((item: {[key: string]: any}) => {
```

Comment:
I tried to type this better like `{[key: string]: Observable<string> | string}` but tsc complain because `string` doesn't have `.pipe` method.